spec/unit/provider/package/dpkg_spec.rb

```diff
@@ -221,19 +221,28 @@ def parser_test(dpkg_output_string, gold_hash, number_of_debug_logs = 0)
       allow(Tempfile).to receive(:new).and_return(tempfile)
     end
 
-    it "installs first if holding" do
+    it "installs first if package is not present and ensure holding" do
+
       allow(provider).to receive(:execute)
+      allow(provider).to receive(:is_package_installed?).and_return(false)
       expect(provider).to receive(:install).once
       provider.hold
     end
 
+    it "skips install new package if package is allready installed" do
+      allow(provider).to receive(:is_package_installed?).and_return(true)
+      expect(provider).to receive(:install).at_most(0).times
```

Comment:
Maybe you can use `not_to`, eg: `expect(provider).not_to receive(:install)`

Reply:
Yeah, the more idiomatic way of doing this with `rspec-mocks` is to use the `not_to` form.

Reply:
@tavilarion @jhelwig done.